document_quick_access_folder_auto_classification/models/document_quick_access_rule.py

```diff
@@ -16,132 +12,44 @@
-                    new_cr.commit()
-            except Exception:
-                if processing_path:
-                    os.unlink(new_element)
-                    new_cr.rollback()
-                raise
-            finally:
-                if processing_path:
-                    new_cr.close()
-            if processing_path:
-                os.unlink(element)
-        return True
-
-    @api.model
-    def _delay_vals(self):
-        return {}
-
-    @api.model
-    def process_document(self, filename, datas):
-        attachment = self.env["ir.attachment"].create(
-            self._get_attachment_vals(filename, datas)
+    def cron_folder_auto_classification(self, limit=None):
+        backends = self.env["edi.backend"].search(
+            [("backend_type_id.code", "=", "document_quick_access")]
         )
-        return attachment._process_quick_access_rules()
-
-    @api.model
-    @job(default_channel="root.document_quick_access_classification")
-    def _process_document(self, element):
-        try:
-            filename = os.path.basename(element)
-            datas = base64.b64encode(open(element, "rb").read())
-            results = self.process_document(filename, datas)
-            return self._postprocess_document(element, results)
-        except OCRException:
-            _logger.warning("Element %s was corrupted" % element)
-            os.unlink(element)
-
-    @api.model
-    def _postprocess_document(self, path, results):
-        filename = os.path.basename(path)
-        if any(result.res_id for result in results):
-            new_path = (
-                self.env["ir.config_parameter"]
-                .sudo()
-                .get_param(
-                    "document_quick_access_auto_classification.ok_path", default=False
-                )
-            )
-        else:
-            new_path = (
-                self.env["ir.config_parameter"]
-                .sudo()
-                .get_param(
-                    "document_quick_access_auto_classification.failure_path",
-                    default=False,
-                )
-            )
-            self.env["document.quick.access.missing"].create(
-                {"name": filename, "attachment_id": results.id}
-            )
-        if new_path:
-            shutil.copy(path, os.path.join(new_path, filename))
-        os.unlink(path)
-        return bool(results)
-
-    def _get_attachment_vals(self, filename, datas):
-        return {"name": filename, "datas": datas}
+        new_limit = limit
+        for backend in backends:
+            new_limit = self._cron_folder_auto_classification(backend, new_limit)
+
+    def _cron_folder_auto_classification_file(self, backend, file_data):
+        exchange_record = backend.create_record(
+            "document_quick_access",
+            {
+                "edi_exchange_state": "input_received",
+                "exchange_file": backend.storage_id.get(file_data, binary=False),
```

Comment:
One issue I found with this is that we cannot use the same folder we were using, as storage is expecting the folder to be inside the odoo path dir.